src/par_iter/from_par_iter.rs

```diff
@@ -1,21 +1,22 @@
-use super::{ParallelIterator, ExactParallelIterator};
+use super::{IntoParallelIterator, ParallelIterator};
 
 use std::collections::{BTreeMap, BTreeSet, HashMap, HashSet};
 use std::hash::{BuildHasher, Hash};
 use std::collections::LinkedList;
 use std::collections::{BinaryHeap, VecDeque};
 
-pub trait FromParallelIterator<PAR_ITER> {
-    fn from_par_iter(par_iter: PAR_ITER) -> Self;
+pub trait FromParallelIterator<ITEM> where ITEM: Send {
```

Comment:
I'm torn about moving `PAR_ITER` to the method. On the one hand, I like the idea that every `collect` can always work from any parallel iterator, and it's certainly true that this can be achieved with buffering. On the other hand, I like the idea that you can control -- for your data structure -- whether you want to expose a buffering mode. 

I guess on balance I like the parameter being on the method. I still suspect we will tinker with this trait a bit, particularly once we have specialization available. (I'd sort of like `.collect()` to ultimately just alias into `.collect_into()`, also.)

Which reminds me, another idea I had been toying with -- basically orthogonal to this PR -- was giving people some control over how `collect()` proceeded (at some point) by having newtype'd wrappers. e.g., for hashmap, you might want to do `let SpareHashMap(map) = ...collect()`, which would be a hint to use folded *hashmaps* (which are faster if there is a lot of overlap). Anyway.